modules/update-restricted-network-cli.adoc

```diff
@@ -0,0 +1,164 @@
+  - mirrors:
+    - <local_registry_host_name>:<local_registry_host_port>/<repository_name>/release
+    source: quay.io/openshift-release-dev/ocp-release
+  - mirrors:
+    - <local_registry_host_name>:<local_registry_host_port>/<repository_name>/release
+    source: quay.io/openshift-release-dev/ocp-v4.0-art-dev
+----
+<1> Specify the entire specification that was showin in the output of the
+`oc adm mirror release` command that mirrored the images for this release.
+.. Create the `ImageContentSourcePolicy` resource that you defined:
++
+----
+$ oc create -f test.yaml
+----
+.. Wait a few minutes, and then confirm that the new `ImageContentSoucePolicy` is in effect by reviewing the node status:
++
+----
+$ oc describe node | grep machineconfig
+                    machineconfiguration.openshift.io/currentConfig: rendered-master-9194b32791755030dcce887c66024113
+                    machineconfiguration.openshift.io/desiredConfig: rendered-master-9194b32791755030dcce887c66024113
+                    machineconfiguration.openshift.io/reason:
+                    machineconfiguration.openshift.io/state: Done
+                    machineconfiguration.openshift.io/currentConfig: rendered-worker-a0f873210ec70fb117835a9c9527db29
+                    machineconfiguration.openshift.io/desiredConfig: rendered-worker-a0f873210ec70fb117835a9c9527db29
+                    machineconfiguration.openshift.io/state: Done
+                    machineconfiguration.openshift.io/currentConfig: rendered-master-9194b32791755030dcce887c66024113
+                    machineconfiguration.openshift.io/desiredConfig: rendered-master-9194b32791755030dcce887c66024113
+                    machineconfiguration.openshift.io/reason:
+                    machineconfiguration.openshift.io/state: Done
+                    machineconfiguration.openshift.io/currentConfig: rendered-worker-a0f873210ec70fb117835a9c9527db29
+                    machineconfiguration.openshift.io/desiredConfig: rendered-worker-a0f873210ec70fb117835a9c9527db29
+                    machineconfiguration.openshift.io/state: Done
+                    machineconfiguration.openshift.io/currentConfig: rendered-worker-a0f873210ec70fb117835a9c9527db29
+                    machineconfiguration.openshift.io/desiredConfig: rendered-worker-a0f873210ec70fb117835a9c9527db29
+                    machineconfiguration.openshift.io/state: Done
+                    machineconfiguration.openshift.io/currentConfig: rendered-master-9194b32791755030dcce887c66024113
+                    machineconfiguration.openshift.io/desiredConfig: rendered-master-9194b32791755030dcce887c66024113
+                    machineconfiguration.openshift.io/state: Done
+
+----
++
+For each machine, the `machineconfiguration.openshift.io/state` must be `Done` before you begin the update.
+
+. Start the cluster update:
++
+----
+$ oc adm upgrade --to-image <local_registry_host_name>:<local_registry_host_port>/ocp/release:<release_version> <1>
```

Comment:
This should be

`oc adm upgrade --force --to-image <local_registry_host_name>:<local_registry_host_port>/ocp-release-dev/ocp-release:<release_version> `

Reply:
@jianlinliu, will you please double-check?

Reply:
since github.com/openshift/oc#111 (4.2 backport), folks should be using `--allow-explicit-upgrade` or `--allow-unsafe-upgrade` instead of a bare `--force`.  But the signature check should only be fatal for by-tag pullspecs like this.  We should be using `@${DIGEST}`, to make sure we're upgrading to the digest we just verified locally.  Once we do that, we may not need any forcing.  Or we may still need `--allow-explicit-upgrade`.

Reply:
@wking, where else do we need to swap to @${DIGEST}?

Reply:
We _could_ use `@${DIGEST}` in the [`--from` argument][1].  That doesn't matter as much, because if you use the current `:${OCP_RELEASE}` and quay.io decides to MitM you (unlikely), then you're still protected by using the digest you verified here, and the cluster just fails the install with "I couldn't find that digest in your mirror, and I couldn't reach the upstream registry to find it there".  That's annoying, but not as annoying as "I'm running a compromised cluster" ;).  And tags are nice and readable, so maybe folks want to pass a by-tag pullspec to `--from` and take that slim denial-of-service risk.  I don't have a preference either way.

[1]: https://github.com/openshift/openshift-docs/pull/16894/files#diff-ae82fa130b6ca45233c1ccc96dce28ebR178

Reply:
Ah, true, if `oc adm upgrade` does use `${DIGEST}`, it is probably safe enough for `oc adm mirror` not to do so. Still, I think it’s a bad habit to use a the tag that may change after verification, even if it is safe in this particular case —  that reference can easily be copy&pasted elsewhere by someone who is not aware of this review discussion and the intricacies of this particular situation.